scripting/src/main/java/org/infinispan/scripting/impl/DataTypedCache.java

```diff
@@ -69,22 +77,27 @@ public V get(Object key) {
 
    @Override
    public CacheSet<K> keySet() {
-      CacheStream<K> stream = getDelegate().keySet().stream().map(this::toDataType);
-      return new SetAsCacheSet<>(stream.collect(Collectors.toSet()));
+      CacheStream<K> stream = getDelegate().keySet().stream().map(
+            new ValueToTypedValueFunction<>(dataTypedCacheManager.dataType, dataTypedCacheManager.marshaller));
+      return new SetAsCacheSet<>(stream.collect(
```

Comment:
I am not familiar with the script cache stuff, but this will delazify the cache contents and hold it all into memory.  Is this okay?